test/Quake/lift_array.qke

```diff
@@ -125,3 +125,50 @@ func.func @test2() -> !quake.veq<2> {
 // GLOBAL-DAG:     cc.global constant private @__nvqpp__mlirgen__function_test_complex_constant_array._Z27test_complex_constant_arrayv.rodata_{{[0-9]+}} (dense<[(0.707106769,0.000000e+00), (0.707106769,0.000000e+00), (0.000000e+00,0.000000e+00), (0.000000e+00,0.000000e+00)]> : tensor<4xcomplex<f32>>) : !cc.array<complex<f32> x 4>
 // GLOBAL-DAG:     cc.global constant private @__nvqpp__mlirgen__function_custom_h_generator_1._Z20custom_h_generator_1v.rodata_{{[0-9]+}} (dense<[(0.70710678118654757,0.000000e+00), (0.70710678118654757,0.000000e+00), (0.70710678118654757,0.000000e+00), (-0.70710678118654757,0.000000e+00)]> : tensor<4xcomplex<f64>>) : !cc.array<complex<f64> x 4>
 // GLOBAL-DAG:     cc.global constant private @test2.rodata_{{[0-9]+}} (dense<[1.000000e+00, 2.000000e+00, 6.000000e+00, 9.000000e+00]>" : tensor<4xf64>) : !cc.array<f64 x 4>
+
+func.func @test_two_stores() {
+  %c0_i64 = arith.constant 0 : i64
+  %c1_i64 = arith.constant 1 : i64
+
+  // qubits = cudaq.qvector(2)
+  %0 = quake.alloca !quake.veq<2>
+
+  // arr1 = [1]
+  %1 = cc.alloca !cc.array<i64 x 1>
+  %2 = cc.cast %1 : (!cc.ptr<!cc.array<i64 x 1>>) -> !cc.ptr<i64>
+  cc.store %c1_i64, %2 : !cc.ptr<i64>
+
+  // t = arr1[0]
+  %3 = cc.load %2 : !cc.ptr<i64>
+
+  // arr2 = [0]
+  %4 = cc.alloca !cc.array<i64 x 1>
+  %5 = cc.cast %4 : (!cc.ptr<!cc.array<i64 x 1>>) -> !cc.ptr<i64>
+  cc.store %c0_i64, %5 : !cc.ptr<i64> // Dominates the next store, don't lift
+
+  // arr2[0] = t
+  cc.store %3, %5 : !cc.ptr<i64>
```

Comment:
The pass should refuse to convert this code because there are 2 stores to the same offset. But I don't see why it wasn't already.

Reply:
second store is not a constant. I will update the fix

Reply:
Updated the fix